### PR TITLE
2024 Importer

### PR DESCRIPTION
**Pull request by awisema**

### CODE REVIEW

File: drivers/hmis_csv_twenty_twenty_four/app/models/hmis_csv_twenty_twenty_four/importer/import_concern.rb [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+    def self.fix_time_columns(row)
+      time_columns.each do |col|
+        row[col] = fix_time_format(row[col])
+      end
+      row
+    end
+
+    # HMIS CSV FORMAT SPECIFICATIONS says under "Data Types"
+    # https://hudhdx.info/Resources/Vendors/HMIS%20CSV%20Specifications%20FY2020%20v1.8.pdf
```
Comment (eanders) [inline drivers/hmis_csv_twenty_twenty_four/app/models/hmis_csv_twenty_twenty_four/importer/import_concern.rb]: This seems a bit outdated now.  I assume the information hasn't changed, maybe we should just remove the link?

File: drivers/hmis_csv_twenty_twenty_four/extensions/grda_warehouse/hud/project_extension.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+              pre_processed_at: Time.current,
+            )
+            new_item = destination_class.new(data)
+            new_item.set_source_hash
+            look_asides << new_item
+          end
+          destination_class.import(look_asides)
+        end
+      end
+    end
```
Comment (eanders) [inline drivers/hmis_csv_twenty_twenty_four/extensions/grda_warehouse/hud/project_extension.rb]: Are these included in the 2022 extension as well?  Should we version them, or maybe remove the old ones, or move them to some shared concern?  Seems odd to inject the same thing multiple times.

File: drivers/hmis_csv_twenty_twenty_four/extensions/grda_warehouse/hud/project_extension.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        copy_data_to_aggregates(enrollments, HmisCsvImporter::Aggregated::Enrollment)
+        copy_data_to_aggregates(exits, HmisCsvImporter::Aggregated::Exit)
+      end
+
+      private def copy_data_to_aggregates(scope, destination_class)
+        pseudo_importer_log = HmisCsvImporter::Importer::ImporterLog.new(id: 0)
+        scope.find_in_batches do |batch|
+          look_asides = []
+          batch.each do |row|
+            data = row.slice(row.class.hmis_structure(version: '2020').keys)
```
Comment (eanders) [inline drivers/hmis_csv_twenty_twenty_four/extensions/grda_warehouse/hud/project_extension.rb]: Does this need updating?
```suggestion
            data = row.slice(row.class.hmis_structure(version: '2024').keys)
```

File: drivers/hmis_csv_twenty_twenty_four/extensions/grda_warehouse/hud/project_extension.rb (hunk 2 above)
Comment (awisema) [reply drivers/hmis_csv_twenty_twenty_four/extensions/grda_warehouse/hud/project_extension.rb]: Yes -- this was specifically for the 2020 -> 2022 migration with clients that were using the versioned aggregation tables, which we removed. Deleted.

**Review by eanders — APPROVED** [review]
@awisema any chance we could add  a smoke test?  It'd be great to take a 2024 file set and confirm that we have the right number of warehouse records after import.

Otherwise, I think this looks great.  Thank you.

File: db/reporting_structure.sql
Comment (eanders) [inline db/reporting_structure.sql:1]: Pretty sure this shouldn't have been included.

**Pull request merged**